### DISCUSSION TITLE
Using firejail from git

### DISCUSSION THREAD

**Discussion by rusty-snake** [netblue30/firejail#4450] [Wiki]
Discussion for discussions about https://github.com/netblue30/firejail/wiki/Using-firejail-from-git.

Predecessor: #3443

**Comment by haplo** [comment]
It'd be great to have a VM-based development workflow. Are there any ongoing work or notes in that regard?

I used [Vagrant](https://www.vagrantup.com) in the past, but I dislike their [license change](https://www.theregister.com/2023/08/11/hashicorp_bsl_licence/). Are there any preferred alternatives?

**Reply by kmk3** [reply]
> It'd be great to have a VM-based development workflow. Are there any ongoing
> work or notes in that regard?

Other than certain hardcoded paths, firejail is designed to be relatively
portable and has little to no dependencies, so any distribution that mostly
follows the Filesystem Hierarchy Standard (FHS, see `hier(7)`) should be fine.
That applies to most distributions, except for the ones that install package
files in a wildly different manner, such as NixOS and GNU Guix SD.

Using a more recent kernel during development would be helpful for working
on/testing more recent features (such as Landlock).  Arch Linux mostly packages
the latest stable upstream versions, so it would be one of the main examples.

Though using older kernels (and program versions) could be helpful to check
that firejail still works properly on them.  Debian would be an example,
especially on older releases that are still supported.

It really depends on what you want to do and on which versions of what programs
are to be tested.

> I used [Vagrant](https://www.vagrantup.com) in the past, but I dislike their
> [license
> change](https://www.theregister.com/2023/08/11/hashicorp_bsl_licence/). Are
> there any preferred alternatives?

It should be fine to just use qemu/virtualbox or whatever VM software you like
as long as it supports whatever features you need.
